libdnf/dnf-sack.cpp

```diff
@@ -2237,21 +2267,50 @@ static void setModuleExcludes(DnfSack *sack, const std::vector<std::string> &hot
 
     dnf_sack_set_module_excludes(sack, excludeQuery.getResultPset());
     dnf_sack_add_module_excludes(sack, excludeProvidesQuery.getResultPset());
+    dnf_sack_set_module_includes(sack, includeQuery.getResultPset());
 }
 
 }
 
-void dnf_sack_filter_modules(DnfSack *sack, GPtrArray *repos, const char *install_root,
-    const char * platformModule)
+std::vector<std::shared_ptr<ModulePackage>> requiresModuleEnablement(DnfSack * sack, const libdnf::PackageSet * installSet)
 {
+    DnfSackPrivate *priv = GET_PRIVATE(sack);
+    if (!priv->module_includes || !priv->moduleContainer) {
+        return {};
+    }
+    libdnf::PackageSet tmp(sack);
+    tmp += *const_cast<libdnf::PackageSet*>(installSet);
+    tmp /= priv->module_includes;
+
+    if (tmp.empty()) {
+        return {};
+    }
+    auto toEnable = priv->moduleContainer->requiresModuleEnablement(*installSet);
+    return toEnable;
+}
+
+void dnf_sack_filter_modules(DnfSack * sack, DnfModulePackageContainer * moduleContainer,
+    const char ** hotfixRepos, const char * install_root, const char * platformModule)
+{
+    if (!install_root) {
+        throw std::runtime_error("Installroot not provided");
+    }
     // TODO: remove hard-coded path
-    g_autofree gchar *defaultsDirPath = g_build_filename(install_root, "/etc/dnf/modules.defaults.d/", NULL);
+    g_autofree gchar *defaultsDirPath = g_build_filename(
+        install_root, "/etc/dnf/modules.defaults.d/", NULL);
     DnfSackPrivate *priv = GET_PRIVATE(sack);
+    if (!moduleContainer) {
+        if (priv->moduleContainer) {
+            delete priv->moduleContainer;
+        }
+        priv->moduleContainer = new ModulePackageContainer(dnf_sack_get_all_arch(sack),
+            install_root, dnf_sack_get_arch(sack));
+        moduleContainer = priv->moduleContainer;
```

Comment:
what's the idea here? That the caller can use their own container instead of sack's one if they want? Still there is no way to access the dnf-sack->moduleContainer, since we are not changing the pointer for the caller.

Reply:
There is problem with DnfSack, PySack, C-Python, and SWIG. In future ModuleContainer will be exclusively in Sack, but that is a future.